tests/test.py

```diff
@@ -0,0 +1,208 @@
+        """Tests adding person who's category is not staff or fellow"""
+        new = self.dojo.add_person("Wekesa','Maina", "Bootcamper", "N")
+        self.assertEqual(new,"Wrong category. Can only be fellow or staff")
+
+    def test_person_added_is_instance_of_person(self):
+        """ Should assert the person added is an instance of Person """
+        self.dojo.add_person("Mengere", "John", "fellow", "Y")
+        self.dojo.add_person("Kipyegon", "Ken", "staff")
+        self.assertIsInstance(self.dojo.fellows[0], Fellow)
+        self.assertIsInstance(self.dojo.staffs[0], Staff)
+
+class TestPrintingRoom(unittest.TestCase):
+    """ Test cases for printing Room and its occupants """
+    def setUp(self):
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["Django"])
+
+    def test_print_room_that_does_not_exist(self):
+        """ Should not print a room that is not in the system """
+        self.assertEqual(self.dojo.print_room('Entebbe'), \
+        "Room Entebbe doesn't exist")
+
+    def test_prints_room_occupants(self):
+        """ Should print all the occupants of a specified room """
+        self.dojo.add_person('Dennis', 'Wachiuri', 'Fellow')
+        self.assertIn("Dennis Wachiuri", self.dojo.print_room('Django'))
+
+    def test_prints_empty_for_unoccupied_room(self):
+        """ Prints null if specified room is empty """
+        self.assertEqual(self.dojo.print_room('Django'), "Django has no occupants")
+
+class TestPrintAllocatedUnallocated(unittest.TestCase):
+    """ Test case for printing allocations """
+    def setUp(self):
+        self.dojo = Dojo()
+        self.dojo.create_room("living", ["Django"])
+        # person 1 to 4 should be allocated to living Room Django
+        self.dojo.add_person('Dennis', 'Person1', 'Fellow' , 'Y')
+        self.dojo.add_person('Dennis', 'Person2', 'Fellow' , 'Y')
+        self.dojo.add_person('Dennis', 'Person3', 'Fellow' , 'Y')
+        self.dojo.add_person('Dennis', 'Person4', 'Fellow' , 'Y')
+
+        # person 5 should be unallocated
+        self.dojo.add_person('Dennis', 'Person5', 'Fellow' , 'Y')
+
+    def test_prints_allocated_successfully(self):
+        allocated = self.dojo.print_allocations("")
+        self.assertIn('Dennis Person4', allocated)
+
+    def test_prints_unallocated_successfully(self):
+        unallocated = self.dojo.print_unallocated("")
+        self.assertIn("Dennis Person5", unallocated)
+
+class TestReallocatePerson(unittest.TestCase):
+    """ Unit tests for reallocating persons """
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def test_rejects_inavalid_id(self):
+        """ Should reject an invalid ID """
+        self.dojo.create_room("Office", "Django")
+        self.dojo.create_room("Office", "Flask")
+        self.dojo.add_person("mwangi", "james", "fellow")
+        james_reallocate = self.dojo.reallocate_person(2, "Flask")
+        self.assertEqual(james_reallocate, "Invalid ID")
```

Comment:
i think an invalid ID could be if someone gives an id of different type other than int. if its a valid id then you can test if it exists